pkg/resourcecollector/resourcecollector.go

```diff
@@ -379,6 +402,106 @@ func (r *ResourceCollector) GetResourcesForType(
 	return objects, pvcsWithOwnerReference, nil
 }
 
+func (r *ResourceCollector) GetResourcesExcludingTypes(
```

Comment:
Should we consider overloading the GetResourcesForType with both the include and exclude parameter?

Reply:
Trying to overload `GetResourcesForType` with excludeType would have been very different code, as the function has already knowledge of for which resource type we have to collect the resources.

Rather I found the excludeType could be easily supported with `GetResources` and again did not want to change the definition of `GetResources` as it is being vendored into different projects. Anyway the common part has been carved into a different function.

Reply:
I am also trying to understand what was the issue with using GetResources and just adding `if !resourceToBeCollectedWithExcludeTypes(resource, groupVersion, crdResources, optionalResourceTypes, excludeResourceTypes)` instead of creating a new function?
Can you point me to what else is it that we need to do different for exclude type in this function?

Reply:
Ok I read your comment about vendoring later..I guess it's ok then...